src/game.js

```diff
@@ -0,0 +1,162 @@
+function Game() {
+  var self = this;
+
+  self.scoreArray = [];
+  self.screenArray = [];
+  self.frameCounter = 0;
+  self.finalScore = 0;
+}
+
+Game.prototype.startGame = function (pinsdropped) {
+  var self = this;
+
+  if (self.scoreArray.length < 21) {
+    self.pinsDroppedEachRoll(pinsdropped);
+
+    for (var i = 1; i < 11; i++) {
+      self.scoreTotal(i);
+    }
+  }
+  return;
+};
+
+Game.prototype.scoreTotal = function (frame) {
+  var self = this;
+  self.finalScore = 0;
+
+    for (var r = 0; r < self.scoreArray.length; r+= 2) {
+      if (self.screenArray[r + 1] === 'X') {
+        self.finalScore += (10 + self.strikeBonus(r));
+
+      } else if (self.screenArray[r + 1] === '/') {
+        self.finalScore += (10 + self.spareBonus(r));
+
+      } else {
+        self.finalScore += self.frameTotal(r);
+      }
+    }
+  return self.finalScore;
+};
+
+Game.prototype.pinsDroppedEachRoll = function (pinsdropped) {
+  var self = this;
+
+  if (self.isIncorrectScore(pinsdropped)) {
+    self.alert();
+  } else {
+    if (self.isStrike(pinsdropped)) {
+
+    } else if (self.isSpare(pinsdropped)) {
+
+    } else {
+      self.scoreArray.push(pinsdropped);
+      self.displayResult(pinsdropped);
+    }
+    self.frameCounter += 1;
+}
+};
+
+
+Game.prototype.isStrike = function (pinsdropped) {
+  var self = this;
+
+  if (self.tenthFrame()) {
+
+    if ( pinsdropped === 10) {
+      self.displayResult('X');
+      self.scoreArray.push(10);
+      return true;
+    }
+
+  } else {
+
+    if ( pinsdropped === 10 && self.frameCounter % 2 === 0 ) {
+      self.displayResult(0);
+      self.displayResult('X');
+      self.scoreArray.push(10);
+      self.scoreArray.push(0);
+      self.frameCounter += 1;
+      return true;
+    }
+  }
+};
+
+
+Game.prototype.isSpare = function (pinsdropped) {
+  var self = this;
+
+  if (self.frameCounter % 2 !== 0) {
+    if (pinsdropped + self.scoreArray[self.frameCounter - 1] === 10) {
+      self.displayResult('/');
+      self.scoreArray.push(pinsdropped);
+      return true;
+    }
+  }
+};
+
+Game.prototype.strikeBonus = function (index) {
+  var self = this;
+
+  if (self.tenthFrame() && self.screenArray[index] === 'X') {
```

Comment:
Whenever we see a big if statement - particularly a nested if statement - we should always be thinking about a refactor. Reasons why this is sub-optimal:

a) It's difficult to read
b) It's difficult to change
c) It's complicated to test all the different outcomes.

I started trying to write an example of how to refactor this, but it was going to be a pretty big endeavour - I think it's worth sitting down and pairing on this at some point during the week.